test/constraint/encoding.test.ts

```diff
@@ -295,6 +296,19 @@ describe('constraints/encoding', () => {
     });
   });
 
+  describe('barMarkWithScaleZero', () => {
```

Comment:
1) See comment about the constraints and move this test accordingly.  
2) You test should **always** cover cases when it returns true and false.  Here you only add one case that cover when it's `false`.   Keep adding test until your constraint have full coverage.  (Use `npm run cover` and look at your constraint after you fix it.)